kornia/augmentation/augmentation.py

```diff
@@ -96,7 +98,7 @@ class RandomHorizontalFlip(AugmentationBase):
     """
 
     def __init__(self, p: float = 0.5, return_transform: bool = False, same_on_batch: bool = False) -> None:
-        super(RandomHorizontalFlip, self).__init__(F.apply_hflip, return_transform)
+        super(RandomHorizontalFlip, self).__init__(F.apply_hflip, F.compute_hflip_transformation, return_transform)
```

Comment:
@shijianjian I think that by including another functional in the base class constructor can overcomplicate the usage of this API.

If we want to let the user to decide wether to return the transform I would do it similar I proposed overloading methods since it will make the API more clean, structured and flexible.

```python
class RandomHorizontalFlip(AugmentationBase):
  def __init__(self, p: float = 0.5, return_transform: bool = False) -> None:
    super(RandomHorizontalFlip, self).__init__(return_transform)

  def get_params(self, input_shape: torch.Size) -> Dict[str, torch.Tensor]:
      # could internally call set self._params
      # and probalby this method name is confusing since it is a generator,
      # not a getter function, probably we should rename.
      return your_generated_params # generate the params as you want

  # OPTIONAL METHOD to compute and return the transform
  def gen_transformation(self, input_shape: torch.Size) -> torch.Tensor:
      # could internally call set self._transform
      # do whatever and
      return your_transform
      # otherwise, we can return by default identity, however we won't have
      # always 2d rigid transforms
      return torch.eye(3)

  def apply(self, input: torch.Tensor) -> torch.Tensor:
     # get your params
     params: Dict[str, torch.Tensor] = self.get_params()
     transform: torch.Tensor = self.get_transform()
     # modify your input and return     
     return output # generate the params as you want
```
The worflow will be:
1 - gen_params()
2 - gen_transformtion()
3 - apply()

Besides, I think we should move all the transformation generators (not random generators) to `kornia.geometry` as in
https://kornia.readthedocs.io/en/latest/geometry.transform.html#kornia.geometry.transform.get_rotation_matrix2d

to enforce kornia core usage from this entry point.

/cc @lferraz what do you think about this proposal ?

Reply:
/cc @ducha-aiki any insights in this ?

Reply:
Actually, I do not think it is a good practice to have ```self._params``` and ```self._transform```. Assigning params to the class instance itself is a bit dangerous I think. I suggest removing those self-assigned parameters.

```python
class RandomHorizontalFlip(AugmentationBase):
  def __init__(self, p: float = 0.5, return_transform: bool = False) -> None:
    super(RandomHorizontalFlip, self).__init__(return_transform)

  def get_params(self, input_shape: torch.Size) -> Dict[str, torch.Tensor]:
      # could internally call set self._params
      # and probalby this method name is confusing since it is a generator,
      # not a getter function, probably we should rename.
      return your_generated_params # generate the params as you want

  # OPTIONAL METHOD to compute and return the transform
  def gen_transformation(self, input_shape: torch.Size, params: Optional[Dict[str, torch.Tensor]]=None) -> torch.Tensor:
      # could internally call set self._transform
      # do whatever and
      return your_transform
      # otherwise, we can return by default identity, however we won't have
      # always 2d rigid transforms
      return torch.eye(3)

  def apply(self, input: torch.Tensor, params: Optional[Dict[str, torch.Tensor]]=None) -> torch.Tensor:
     # get your params
     params: Dict[str, torch.Tensor] = self.get_params()
     transform: torch.Tensor = self.get_transform()
     # modify your input and return     
     return output # generate the params as you want
```